### DISCUSSION TITLE
Redirect of the logout to home page

### DISCUSSION THREAD

**Discussion by Jijiri89** [filamentphp/filament#6916] [General]
Please, how do I redirect my logout to home page of my front end? I currently, if I log out from admin dashboard, it redirect me to login page of the filament admin login. I want a way to redirect it to my welcome page.

**Comment by pboivin** [comment]
I don't think there's a simple solution like a config but you can probably extend the `LogoutResponse` class and bind it in the service container :
https://github.com/filamentphp/filament/blob/2.x/packages/admin/src/Http/Responses/Auth/LogoutResponse.php

In your `AppServiceProvider`:
```php
use App\Filament\MyLogoutResponse;
use Filament\Http\Responses\Auth\Contracts\LogoutResponse as LogoutResponseContract;

// ...

public function boot()
{
    $this->app->bind(LogoutResponseContract::class, MyLogoutResponse::class);
}
```

**Comment by Jijiri89** [comment]
Thanks for your response.
When I put this code into my app Service provider and clicked to logout
from admin dashboard, I get error that, the target class App\filament
\MylogoutResponse does not exist. Please do I need to do anything again?

**Reply by RX-77IT** [reply]
For me it's working: you probably just need to write the correct MyLogoutResponse class (based on the class in the github link above) and put it in the app/Filament folder.

Here's how I wrote mine.
```php
<?php

namespace App\Filament;

use Filament\Http\Responses\Auth\Contracts\LogoutResponse as Responsable;
use Illuminate\Http\RedirectResponse;

class MyLogoutResponse implements Responsable
{
    public function toResponse($request): RedirectResponse
    {
        return redirect()->to(
            route('home'), // put the name of your home route here
        );
    }
}

```

**Comment by Jijiri89** [comment]
That worked! Please are you referring to filament in the confiq folder?
(App/Confiq /filament.php)
Or above method need a controller?

**Reply by RX-77IT** [reply]
I was referring to pboivin's link in his post ([https://github.com/filamentphp/filament/blob/2.x/packages/admin/src/Http/Responses/Auth/LogoutResponse.php](https://github.com/filamentphp/filament/blob/2.x/packages/admin/src/Http/Responses/Auth/LogoutResponse.php)) to use as a base to write the MyLogoutResponse class.

If it's working, I don't think you need anything else.

**Comment by Jijiri89** [comment]
Thanks, I am a bit confused of where exactly to put those code.

**Comment by Jijiri89** [comment]
Hello boss,
Please, I tried implementing this filament logout redirect to home page but
I still have challenges in the implementation.

My problem is that, do I need to create a controller for this

<?php


namespace App\Filament;


use Filament\Http\Responses\Auth\Contracts\LogoutResponse as Responsable;

use Illuminate\Http\RedirectResponse;


class MyLogoutResponse implements Responsable

{

    public function toResponse($request): RedirectResponse

    {

        return redirect()->to(

            route('home'), // put the name of your home route here

        );

    }

}


Or where exactly in the laravel will l define that logic?

I created a controller where I copied and pasted this codes but it didn't
work.

So I am confused of where to go. Kindly pardon me for my plenty questions.

**Reply by pboivin** [reply]
Hi @Jijiri89, 

Sorry you're still having issues with this. The code @RX-77IT shared should go in a file at this location: `app/Filament/MyLogoutResponse.php`. No need to add a controller or anything else.

You will also need to add some more code in `app/Providers/AppServiceProvider.php`, like I mentioned in my first comment.

If this is still a bit confusing to you, I think you will need to spend a bit more time in the Laravel documentation. Here are some links that can be useful:

- https://laravel.com/docs/10.x/providers
- https://laravel.com/docs/10.x/container#binding
- https://www.php.net/manual/en/language.oop5.interfaces.php

**Comment by Jijiri89** [comment]
Thanks, I am most grateful sir.
So this means I have to manually create that file or I have to use artisan
command to create that file into: app/Filament/MylogoutResponse.php?

**Comment by Jijiri89** [comment]
Please, I searched for app/filament /MylogoutResponse directory but I can't
find it in my project

**Comment by Jijiri89** [comment]
Please, I searched for app/filament /MylogoutResponse directory but I can't
find it in my project

**Comment by Jijiri89** [comment]
Thanks so much brother for your kind help. It has worked now. I am most
grateful
